### ISSUE TITLE
Why does SAMLEntryPoint require an instance of MetadataManager?

### ISSUE THREAD

**Issue by kristofferpeterhansel** [spring-projects/spring-security-saml#147]
So I am trying to use this library for SAML SSO in a solution. But since we are a multi-tenant solution we need to be a lot more dynamic in how we load metadata. So I am trying to replace that.

But SAMLEntryPoint requires an instance of MetadataManager to be used. And as far as I can tell there is no reason for it anymore. It's not referenced directly anywhere in the code - outside of the setter and the check to see if it has been set.

**Comment by vschafer** [comment]
Fair point, I'll take it out, no need to keep it around. Contributions on dynamic metadata reloading are welcome, never had time to improve it further.

**Comment by kristofferpeterhansel** [comment]
We'll see if I come up with something that can easily be integrated back into this project. Still wrapping my head around this thing while moving from super simple prototype to something that will actually work in our setup

**Comment by kristofferpeterhansel** [comment]
I may have been caught up in one of those moments where Spring tells me one specific thing isn't working and you sort of presume that is the only thing you need to fix. After doing a quick-fix I ended up with other parts of this framework that requires an MetadataManager instance. Making it very hard to replace it.

The issue is still valid though. It's just not going to help me very much to fix it in that one place

**Comment by karanb192** [comment]
@kristofferpeterhansel  Were you able to configure spring to use multi tenant SAML authentication?

**Comment by kristofferpeterhansel** [comment]
> @kristofferpeterhansel Were you able to configure spring to use multi tenant SAML authentication?
No. We ended up with a more direct SAML use of SAML for what we needed. It's been a while since I worked in company where we had that issue though

**Comment by kristofferpeterhansel** [comment]
@karanb192 I can't at this point remember exactly what I ended up doing. I think I cobbled something simple and functional directly using OpenSAML.
